contributors/design-proposals/sig-cli/apply_v2.md

```diff
@@ -0,0 +1,201 @@
+# Apply v2
+
+## Background
+
+`kubectl apply` reads a file or set of files, and updates the cluster state based off the file contents.
+It does a couple things:
+
+1. Create / Update / (Delete) the resource based on the resources
+2. Only set / delete fields defined in the file either currently or in during last update from the file
+  - Retain updates to fields made by controllers that were never managed in the file
+3. Merge fields based on the field name, merge lists based on a field contained within the list elements
+
+The apply logic has been difficult to maintain from a contributor perspective, and difficult
+to understand from a user perspective.  Much of this pain has comes from the `apply` command out
```

Comment:
Much of this pain has come from trying to paper over API schema complexity in an ad hoc incremental fashion using imperative patch directives.